yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/configs/ConfigWithOptions.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * ConfigWithOptions.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.relational.yamltests.configs;
+
+import com.apple.foundationdb.relational.yamltests.YamlRunner;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An implementation of {@link YamlTestConfig} that sets additional options on top of a base config.
+ */
+public class ConfigWithOptions implements YamlTestConfig {
```

Comment:
This is a nit (I don't think it should prevent us from merging this PR) but it would seem to be a simpler arrangement if the options were part of the base implementation of the `Config` rather than a delegation. This way you could create a `new MultiServerConfig(0, 1113, 1114, Map.of(Option, value)))`, and the test in YamlTestConfigExclusions would be `config instanceof JDBCInProcessConfig && config.getOptions(Option)`

Reply:
That is true, but then you wouldn't be able to easily build an embedded, or combine different sets of options. 
There's actually a part of me that just wanted to make the fact that it is JDBC visible in the options, or make it a method such that `ConfigWithOptions` has `isJDBC()` which calls `underlying.isJDBC`, but I didn't feel like trying to engineer that at this point was the right call given that we have only the two things.